### ISSUE TITLE
Slope always converts Route strings to Route RegExps

### ISSUE THREAD

**Issue by wibbuffey** [wibbuffey/slope#12]
**Describe the bug**  
`Slope.Server.run()` will _always_ convert `String`s to `RegExp`s when checking if a route exists. This means that it only checks if the URL contains a route, not _is_ a route.

**To Reproduce**  
```javascript
// imports, etc
const Slope = require("slope");
const server = new Slope.Server();

server.routes.add("/", (Client, Result) => {
  Result.send(`Received request: ${Client.url}`);
});

server.run();

// If you visit any URL now, because it contains "/",
// slope will show this webpage, despite it not using
// the wildcard operator ("*").
```

**Expected behavior**  
Obviously, slope should only display the webpage when `/` is visited. 

**Environment**  
 - OS: Windows 10 *(only for now!)*
 - Node.js Version: 16
 - Slope Version: 1.1.3

**Additional context**  
This bug affects practically every part of Slope code, as usually `/` is added **FIRST**, meaning that it will override all behavior. This also means the example in the README, which prides itself on being "working slope code", is broken (as of v1.1.0).

**Comment by wibbuffey** [comment]
I'm noting here that this was actually intended behavior, so if in 1.1.0 if you used `"/my/url/.+"` instead of treating it as a literal string, it would be viewed as a `RegExp`. However, in hindsight, this was a _really_ bad idea: how could slope tell the difference between `/my/url/.+` and `"/my/url/.+"`?

My initial "fix" was to only convert strings to `RegExp`s when they ended with a slash, but eventually I reverted this change (in my own local clone) as slashes at the end of URLs are not uncommon (and I actually use them sometimes!) and slope, just like before, can't tell the difference.

So I'm now working a patch that _doesn't_ convert `RegExp`s to strings. Once I push it, within a few hours, 1.2.0 will be done and published. Hopefully this approach will (finally) fix this bug.

**Comment by wibbuffey** [comment]
Fixed(?) in 2f761f3.

**Comment by wibbuffey** [comment]
Extra bugs relating to the fix were fixed in b3c1bb0.

**Comment by wibbuffey** [comment]
Not fixed -- JavaScript converts RegExps to Strings when used as Object keys. Working on a fix that involves a new `Buffer` type.
